src/session/session_compact.c

```diff
@@ -360,6 +360,31 @@ __compact_worker(WT_SESSION_IMPL *session)
     return (ret);
 }
 
+/*
+ * __wt_compact_check_eligibility --
+ *     Function to check whether the specified URI is eligible for compaction.
+ */
+int
+__wt_compact_check_eligibility(WT_SESSION_IMPL *session, const char *uri)
+{
+    bool is_server;
+
+    is_server = session == S2C(session)->background_compact.session;
+
+    /* The background compaction server should not compact the history store file. */
+    if (is_server && WT_STREQ(uri, WT_HS_URI))
```

Comment:
I feel it is better to make the code simpler by not allowing compact on the history store for the foreground compact also.

Reply:
@kommiharibabu and I discussed this a bit more. There are a couple of points to address:

- Consistency
Currently, it is possible to compact the HS through foreground compaction. In order to make the code easier to read and more consistent, it would be great not to make it different for the background compaction.

- Disruption
Compacting the HS can disrupt the workload like any other table. This one is trickier as it is a shared table, that's why we first wanted the background compaction to avoid that table. However, with WT-11343, we should see that there is no work to be done when analyzing the HS and we would skip compaction on it.

See the last commit: 4dbee735c39da2b996de7ac82e5b59fe562bc66e.

cc: @keitharnoldsmith @Sean04